datasource/src/main/scala/quasar/physical/s3/AsyncHttpClientBuilder.scala

```diff
@@ -52,16 +56,31 @@ object AsyncHttpClientBuilder extends Http4sClientBuilder {
         s"http4s-async-http-client-worker-${i}"
       })).build()
 
+  private[s3] def sortProxies(proxies: List[Proxy]): List[Proxy] =
+    proxies.sortWith((l, r) => (l.`type`, r.`type`) match {
+      case (ProxyType.HTTP, ProxyType.DIRECT) => true
+      case (ProxyType.SOCKS, ProxyType.DIRECT) => true
+      case _ => false
```

Comment:
I think there has to be one more case set to true according to the contract? Either HTTP < SOCKS or the other way around. 

I don't mind sorting that way in this case, but usually I prefer to map to int and then use `sortBy` - that scales better when the amount of elements grows.

Reply:
>I think there has to be one more case set to true according to the contract?

Not necessarily. [`sortWith`'s documentation indicates it implements a stable sort](https://www.scala-lang.org/api/current/scala/collection/immutable/List.html#sortWith(lt:(A,A)=>Boolean):C), so elements deemed "equal" will appear in the same order in the resulting list. I'd rather not modify the order `SOCKS` and `HTTP`.

Reply:
Ah yes, sorry - you are right.